File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int kthSmallest(TreeNode root, int k) {
        int[] res = new int[2];

        trav(root, k, res);

        return res[1];
    }

    void trav(TreeNode root, int k, int[] res) {
        if (root == null) return;

        trav(root.left, k, res);
        res[0] += 1;
        if (res[0] == k) {
            System.out.println("done for " + root.val);
            res[1] = root.val;
            return;
        }
        trav(root.right, k, res);

        return;
    }

    
}

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func inorderTraversal(root *TreeNode) []int {
    res := []int{}

    addInOrder(&res, root)
    return res
}

func addInOrder(res *[]int, root *TreeNode) {
    if root == nil {
        return
    }

    addInOrder(res, root.Left)
    *res = append(*res, root.Val)
    addInOrder(res, root.Right)
}
